src/Solvers/plan_transforms.jl

```diff
@@ -47,17 +41,27 @@ function plan_backward_transform(A::CuArray, topo, dims, planner_flag)
     return CUDA.CUFFT.plan_ifft!(A, dims)
 end
 
-forward_orders(::Type{Periodic}, ::Type{Bounded} , ::Type{Bounded})  = (3, 2, 1)
-forward_orders(::Type{Periodic}, ::Type{Bounded} , ::Type{Periodic}) = (2, 1, 3)
-forward_orders(::Type{Bounded} , ::Type{Periodic}, ::Type{Bounded})  = (1, 3, 2)
-forward_orders(::Type{Bounded} , ::Type{Bounded} , ::Type{Periodic}) = (1, 2, 3)
-forward_orders(::Type{Bounded} , ::Type{Bounded} , ::Type{Bounded})  = (1, 2, 3)
+plan_backward_transform(A::Union{Array, CuArray}, ::Flat, args...) = nothing
+plan_forward_transform(A::Union{Array, CuArray}, ::Flat, args...) = nothing
+
+batchable_GPU_topologies = ((Periodic, Periodic, Periodic),
+                            (Periodic, Periodic, Bounded),
+                            (Bounded, Periodic, Periodic))
+
+const PeriodicOrFlatType = Union{Type{Periodic}, Type{Flat}}
+const BoundedOrFlatType = Union{Type{Periodic}, Type{Flat}}
 
-backward_orders(::Type{Periodic}, ::Type{Bounded} , ::Type{Bounded})  = (1, 2, 3)
-backward_orders(::Type{Periodic}, ::Type{Bounded} , ::Type{Periodic}) = (3, 1, 2)
-backward_orders(::Type{Bounded} , ::Type{Periodic}, ::Type{Bounded})  = (2, 1, 3)
-backward_orders(::Type{Bounded} , ::Type{Bounded} , ::Type{Periodic}) = (3, 1, 2)
-backward_orders(::Type{Bounded} , ::Type{Bounded} , ::Type{Bounded})  = (1, 2, 3)
+forward_orders(::PeriodicOrFlatType, ::BoundedOrFlatType,  ::BoundedOrFlatType)  = (3, 2, 1)
```

Comment:
What's the principle in picking the forward and backward orders?  I'm guessing it's complicated and I don't doubt that you did it correctly, I'm just curious to better understand at least a little how this is picked.

Reply:
Ah in principle the order does not matter of course but in practice we want to perform the `Bounded` forward transforms first because on the GPU we take the real part after a forward transform, so if you did the `Periodic` transform first you would lose the information in the imaginary components after a `Bounded` forward transform.

For the same reason, `Bounded` backward transforms are applied after `Periodic` backward transforms.

Hmmm, I'll add a suggestion that adds some comments that say this.

Reply:
I had no idea that GPUs did that so thanks for the explanation.  I'm happy with the answer and expect that anyone looking at this bit of code should know what they're doing before they do any modifications.